apps/dashboard/src/app/(app)/team/[team_slug]/[project_slug]/insight/webhooks/components/CreateWebhookModal.tsx

```diff
@@ -0,0 +1,380 @@
+        setCurrentStep(WebhookFormSteps.FilterDetails);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleOpenModal = () => {
+    formHook.reset();
+    setCurrentStep(WebhookFormSteps.BasicInfo);
+    setIsOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsOpen(false);
+  };
+
+  const { testWebhookEndpoint, isTestingMap, testResults, isRecentResult } =
+    useTestWebhook(clientId);
+
+  const handleTestWebhook = async (webhookUrl: string) => {
+    const filterType = formHook.getValues("filterType");
+    await testWebhookEndpoint(webhookUrl, filterType);
+  };
+
+  return (
+    <Dialog open={isOpen}>
+      <DialogTrigger asChild>
+        <Button onClick={handleOpenModal}>New Webhook</Button>
+      </DialogTrigger>
+
+      <DialogContent
+        dialogCloseClassName="hidden"
+        className="max-h-[85vh] max-w-3xl overflow-y-auto"
+      >
+        <div className="flex items-center justify-between">
+          <DialogTitle className="font-medium text-lg">
+            Create New Webhook
+          </DialogTitle>
+          <Button
+            variant="ghost"
+            size="icon"
+            onClick={handleCloseModal}
+            className="h-6 w-6"
+          >
+            <XIcon className="h-4 w-4" />
+            <span className="sr-only">Close</span>
+          </Button>
+        </div>
+
+        <Form {...formHook}>
+          <form
+            onSubmit={formHook.handleSubmit(handleSubmit)}
+            className="space-y-4"
+          >
+            {/* Step indicator */}
+            <StepIndicator currentStep={currentStep} />
+
+            {/* Step Content */}
+            <div className="min-h-[300px]">
+              {currentStep === WebhookFormSteps.BasicInfo && (
+                <BasicInfoStep form={formHook} />
+              )}
+
+              {currentStep === WebhookFormSteps.FilterDetails && (
+                <FilterDetailsStep
+                  form={formHook}
+                  eventSignatures={eventAbi.signatures}
+                  functionSignatures={txAbi.signatures}
+                  fetchedAbis={eventAbi.fetchedAbis}
+                  abiErrors={{}}
+                  fetchedTxAbis={txAbi.fetchedAbis}
+                  txAbiErrors={{}}
+                  isFetchingEventAbi={eventAbi.isFetching}
+                  isFetchingTxAbi={txAbi.isFetching}
+                />
+              )}
+
+              {currentStep === WebhookFormSteps.Review && (
+                <ReviewStep
+                  form={formHook}
+                  isTestingMap={isTestingMap}
+                  testResults={testResults}
+                  isRecentResult={isRecentResult}
+                  testWebhookEndpoint={handleTestWebhook}
+                />
+              )}
+            </div>
+
+            {/* Navigation Buttons */}
+            <div className="flex justify-between pt-4">
+              {currentStep !== WebhookFormSteps.BasicInfo ? (
```

Comment:
The navigation buttons should also go inside the step component so this is easier to maintain and update later